src/components/Form.jsx

```diff
@@ -12,24 +13,126 @@ class Form extends Component {
             knowProb: 'No',
             clientQuestion: '',
-            problemDesc: ''
+            problemDesc: '',
+            clientZip: '',
+            clientCare: '',
+            clientPay: '',
+            anotherProb: 'No',
+            waterTesting: 'No',
+            formErrors: {
+                clientName: '',
+                clientEmail: '',
+                clientHouseAge: '',
+                clientAddress: '',
+                knowProb: '',
+                clientQuestion: '',
+                problemDesc: '',
+                clientZip: '',
+                clientCare: '',
+                clientPay: '',
+                anotherProb: '',
+                waterTesting: ''
+            },
+            clientNameValid: '',
+            clientEmailValid: '',
+            clientHouseAgeValid: '',
+            clientAddressValid: '',
+            clientKnowProbValid: '',
+            clientQuestionValid: '',
+            clientProblemDescValid: '',
+            clientZipValid: '',
+            clientCareValid: '',
+            clientPayValid: '',
+            anotherProbValid: '',
+            waterTestValid: ''
           };
         this.handleChange=this.handleChange.bind(this);
         this.handleSubmit=this.handleSubmit.bind(this);
     }
 
+    validateField(fieldName, value) {
+        let fieldValidationErrors = this.state.formErrors;
+        let clientNameValid = this.state.clientNameValid;
+        let clientEmailValid = this.state.clientEmailValid;
+        let clientHouseAgeValid = this.state.clientHouseAgeValid;
+        let clientAddressValid = this.state.clientAddressValid;
+        let clientKnowProbValid = this.state.clientKnowProbValid;
+        let clientQuestionValid = this.state.clientQuestion;
+        let clientProblemDescValid = this.state.clientProblemDescValid;
+        let clientZipValid = this.state.clientZipValid;
+        let clientCareValid = this.state.clientCareValid;
+        let clientPayValid = this.state.clientPayValid;
+        let anotherProbValid = this.state.anotherProbValid;
+        let waterTestValid = this.state.waterTestValid;
+        console.log('fieldName', fieldName);
+        switch(fieldName) {
+            case 'clientName': {
+                clientNameValid = value.length > 0;
+                console.log('clientNameValid: ', clientNameValid);
+                fieldValidationErrors.clientName = clientNameValid ? '' : 'Please provide a name';
+                break;
+            }
+            case 'clientEmail': {
+                clientEmailValid = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(value);
+                console.log('clientEmailValid: ', clientEmailValid);
+                fieldValidationErrors.clientEmail = clientEmailValid ? '' : 'Email is invalid';
+                break;
+            }
+            case 'clientHouseAge': {
+                console.log('house age length', value.length);
+                clientHouseAgeValid = value.length > 0;
```

Comment:
Similar to your regex for client email I would test if house age only contains numbers I would do the same for zipcode